src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,47 @@
+package duke.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a deadline task.
+ */
+public class Deadline extends Task {
+
+    private String by;
+    private LocalDateTime time;
+
+    /**
+     * Constructs a deadline Task.
+     * @param description description of the task
+     * @param by deadline of the task
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+            time = LocalDateTime.parse(by, formatter);
+        } catch (Exception err) {
```

Comment:
Might it be better to catch the specific type of exception thrown, rather than Exceptions in general?

Reply:
Indeed, the `|` operator can be used to catch multiple exceptions, as in
```suggestion
        } catch (IllegalArgumentException | DateTimeParseException e) {
```